test/test_invalid_headers.py

```diff
@@ -365,6 +367,27 @@ def test_push_promise_skip_normalization(self, frame_factory, headers):
         )
         assert c.data_to_send() == pp_frame.serialize()
 
+    @pytest.mark.parametrize('headers', strippable_header_blocks)
+    def test_strippable_headers(self, frame_factory, headers):
+        """
+        Test connection related headers are removed before sending.
+        """
+        c = h2.connection.H2Connection()
+        c.initiate_connection()
+
+        # Clear the data, then try to send headers.
+        c.clear_outbound_data_buffer()
+        c.send_headers(1, headers)
+
+        conn_headers = (
+            'connection', 'proxy-connection', 'keep-alive',
+            'transfer-encoding', 'upgrade'
+        )
+        stripped_headers = [h for h in headers if h[0] not in conn_headers]
```

Comment:
This replicates too much of the logic of the underlying code. In this case, we can say without loss of generality that the end result should be `stripped_headers == base_request_headers``, so let's use that as our assertion instead.